docs/building_blocks_for_flexible_content.md

```diff
@@ -0,0 +1,566 @@
+
+##### Examples
+
+```yaml
+- type: govspeak
+  content: |
+    <h3>Lorem heading</h3>
+    <p>Korem ipsum dolor sit amet, consectetur adipiscing elit. Etiam eu turpis
+    molestie, dictum esta, mattis tellus.</p>
+```
+
+```yaml
+- type: govspeak
+  content:
+    content-type: text/govspeak
+    content: |
+      ### Lorem heading
+
+      Korem ipsum dolor sit amet, consectetur adipiscing elit. Etiam eu turpis
+      molestie, dictum esta, mattis tellus.
+```
+
+#### Heading
+
+A wrapper around the [Heading component](https://components.publishing.service.gov.uk/component-guide/heading). Note, though, that the block uses `content` where the component uses `text` for the title text key. This is so that headings can be indexed (only values inside a `content` key will be indexed when being published from Whitehall - see [Indexing block content in search](#indexing-block-content-in-search))
+
+##### Example
+
+```yaml
+- type: heading
+  content: Porem ipsum dolor
+```
+
+#### Image
+
+A simple image.
+
+##### Example
+
+```yaml
+- type:
+  image:
+    alt: "Placeholder alt text"
+    sources:
+      desktop: "landing_page/placeholder/desktop.png"
+      desktop_2x: "landing_page/placeholder/desktop_2x.png"
+      mobile: "landing_page/placeholder/mobile.png"
+      mobile_2x: "landing_page/placeholder/mobile_2x.png"
+      tablet: "landing_page/placeholder/tablet.png"
+      tablet_2x: "landing_page/placeholder/tablet_2x.png"
+```
+
+#### Main Navigation
+
+The landing-page top-level navigation. It supports a row of links, and sub-menus that will only appear on mobile (where the entire menu is compressed into a single drop-down). The only supported keys are the `type` key and the `navigation_group_id` key, which should point to an id of a [navigation group](#navigation-groups) to use for the menu items. Note that the styling of the first link is different, and it is assumed to be the homepage in a group of landing pages.
+
+##### Example
+
+```yaml
+- type: main_navigation
+  navigation_group_id: Top Menu
+```
+
+#### Quote
+
+A blockquote.
+
+##### Example
+
+```yaml
+- type: quote
+  text: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc vulputate libero et velit interdum, ac aliquet odio mattis class"
+  cite: "Lorem ipsum dolor sit, 28 September 2024"
+```
+
+#### Share links
+
+A wrapper around a [Heading component](https://components.publishing.service.gov.uk/component-guide/heading) followed by a [Share links component](https://components.publishing.service.gov.uk/component-guide/share_links)
+
+##### Example
+
+```yaml
+- type: share_links
+  links:
+    - href: /twitter-profile
```

Comment:
TO DO: check the YAML indentation in all of the examples as they are likely to be copy and pasted into Whitehall.